### PR TITLE
add dygraph multiplex

### PR DESCRIPTION
**Pull request by hutuxian**
### PR types

Function optimization
### PR changes

OPs
### Describe

Add dygraph implementation for multiplex op

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -5757,7 +5757,7 @@ def multiplex(inputs, index):
 
     Args:
        inputs (list): The input Tensor list. The list elements are N-D Tensors of data types float32, float64, int32, int64. All input Tensor shapes should be the same and rank must be at least 2.
-       index (Variable): Used to select some rows in the input Tensor to construct an index of the output Tensor. It is a 2-D Tensor with data type int32 or int64 and shape [M, 1], where M is the number of input Tensors.
+       index (Tensor): Used to select some rows in the input Tensor to construct an index of the output Tensor. It is a 2-D Tensor with data type int32 or int64 and shape [M, 1], where M is the number of input Tensors.
 
     Returns:
         Variable(Tensor): Output of multiplex OP, with data type being float32, float64, int32, int64.
```
Comment (TCChenlong) [inline python/paddle/fluid/layers/nn.py]: Returns:
        Tensor: Output of multiplex OP

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -5757,7 +5757,7 @@ def multiplex(inputs, index):
 
     Args:
        inputs (list): The input Tensor list. The list elements are N-D Tensors of data types float32, float64, int32, int64. All input Tensor shapes should be the same and rank must be at least 2.
-       index (Variable): Used to select some rows in the input Tensor to construct an index of the output Tensor. It is a 2-D Tensor with data type int32 or int64 and shape [M, 1], where M is the number of input Tensors.
+       index (Tensor): Used to select some rows in the input Tensor to construct an index of the output Tensor. It is a 2-D Tensor with data type int32 or int64 and shape [M, 1], where M is the number of input Tensors.
```
Comment (TCChenlong) [inline python/paddle/fluid/layers/nn.py]: 少了name

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (hutuxian) [reply python/paddle/fluid/layers/nn.py]: done

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (hutuxian) [reply python/paddle/fluid/layers/nn.py]: donne

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]
LGTM

**Pull request merged**